core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -475,7 +475,8 @@ public int register(String subject,
           ParsedSchema undeletedSchema = parseSchema(getSchemaEntityFromSchemaValue(schemaValue));
           if (parsedSchema.references().isEmpty()
               && !undeletedSchema.references().isEmpty()
-              && parsedSchema.deepEquals(undeletedSchema)) {
+              && parsedSchema.deepEquals(undeletedSchema)
+              && (schemaId < 0 || schemaId == schemaValue.getId())) {
```

Comment:
how to understand the schemId < 0 here?

Reply:
When SR is not in IMPORT mode, a new schema gets id -1 to indicate that an id needs to be generated.  During IMPORT mode, a user is allowed to set the schema id to a non-negative number.